sdk/rust/oak_utils/src/tests.rs

```diff
@@ -22,72 +22,165 @@ use std::fs::File;
+            new_content: None,
+        }
+        .action(),
+        FileAction::Remove
+    );
+    assert_eq!(
+        // `mod.rs` is treated as an exception because it's not generated by
+        // `protoc_rust` and `protoc_rust_grpc`.
+        FileDiff {
+            filename: "mod.rs".to_string(),
+            old_content: Some("1".to_string()),
+            new_content: None,
+        }
+        .action(),
+        FileAction::Ignore
+    );
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: None,
+            new_content: Some("2".to_string()),
+        }
+        .action(),
+        FileAction::Update {
+            new_content: "2".to_string()
+        }
+    );
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: Some("".to_string()),
+            new_content: None,
+        }
+        .action(),
+        FileAction::Remove
+    );
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: None,
+            new_content: Some("".to_string()),
+        }
+        .action(),
+        FileAction::Update {
+            new_content: "".to_string()
+        }
+    );
 }
 
 #[test]
-fn get_changed_and_removed_files_test() {
+fn get_file_diffs_test() {
     let old_files = hashmap! {
-        "1" => "string1",
-        "2" => "string2",
-        "3" => "string3",
+        "1".to_string() => "string1".to_string(),
+        "2".to_string() => "string2".to_string(),
+        "3".to_string() => "string3".to_string(),
     };
     let new_files = hashmap! {
-        "1" => "changed_string1",
-        "3" => "string3",
-        "4" => "string4",
+        "1".to_string() => "changed_string1".to_string(),
+        "3".to_string() => "string3".to_string(),
+        "4".to_string() => "string4".to_string(),
     };
-    let expected_changed_files: &[&str] = &["1", "4"];
-    let expected_removed_files: &[&str] = &["2"];
+    let mut expected_file_diffs = vec![
+        FileDiff {
+            filename: "1".to_string(),
+            old_content: Some("string1".to_string()),
+            new_content: Some("changed_string1".to_string()),
+        },
+        FileDiff {
+            filename: "2".to_string(),
+            old_content: Some("string2".to_string()),
+            new_content: None,
+        },
+        FileDiff {
+            filename: "3".to_string(),
+            old_content: Some("string3".to_string()),
+            new_content: Some("string3".to_string()),
+        },
+        FileDiff {
+            filename: "4".to_string(),
+            old_content: None,
+            new_content: Some("string4".to_string()),
+        },
+    ];
+    expected_file_diffs.sort();
 
-    let old_temp_dir = tempfile::tempdir().unwrap();
-    let new_temp_dir = tempfile::tempdir().unwrap();
+    let mut file_diffs = get_file_diffs(&old_files, &new_files);
+    file_diffs.sort();
```

Comment:
Alternatively could use a hash set, and avoid sorting (though probably need to implement / derive hashing instead of ord). Up to you.

Reply:
Done